src/Compilation.zig

```diff
@@ -5922,13 +5922,13 @@ pub fn addCCArgs(
                         // function was called.
                         try argv.append("-fno-sanitize=function");
                     }
-
-                    if (comp.config.san_cov_trace_pc_guard) {
-                        try argv.appendSlice(&.{ "-Xclang", "-fsanitize-coverage-trace-pc-guard" });
-                    }
                 }
             }
 
+            if (comp.config.san_cov_trace_pc_guard) {
+                try argv.append("-fsanitize-coverage=trace-pc-guard");
+            }
+
```

Comment:
Seems like this should still be in the block above (which is just there for grouping purposes).